GAE/src/org/waterforpeople/mapping/dataexport/SurveySummaryExporter.java

```diff
@@ -213,17 +213,27 @@ protected SummaryModel buildDataModel(String surveyId, String serverBase, String
 
     }
 
+    /**
+     * loads just enough question data to generate the simplest report
+     * @param surveyId
+     * @param performRollups
+     * @param serverBase
+     * @param apiKey
+     * @return
+     * @throws Exception
+     */
     protected Map<QuestionGroupDto, List<QuestionDto>> loadAllQuestions(
             String surveyId, boolean performRollups, String serverBase, String apiKey)
             throws Exception {
         Map<QuestionGroupDto, List<QuestionDto>> questionMap = new HashMap<QuestionGroupDto, List<QuestionDto>>();
+        //TODO: fetching in nested loops is inefficient, but we need the ordering of groups and questions in them 
         orderedGroupList = fetchQuestionGroups(serverBase, surveyId, apiKey);
         rollupOrder = new ArrayList<QuestionDto>();
         for (QuestionGroupDto group : orderedGroupList) {
```

Comment:
Even though we need the question order, its still probably better to fetch the questions themselves in one go and then do the local processing to reorder them based on their question group.